libs/messaging/db-message-bus-impl/src/main/kotlin/net/corda/messagebus/db/consumer/DBCordaConsumerImpl.kt

```
@@ -176,9 +176,12 @@ internal class DBCordaConsumerImpl<K : Any, V : Any> constructor(
                 deserializeValue(dbRecord.value),
                 dbRecord.timestamp.toEpochMilli()
             )
+        }.filter {
+            it.value != null || it.topic.endsWith(".dlq")
```

Is there a sensible place to add `".dlq"` as a constant?

---

I don't think that business logic here should depend on the queue name. If such a feature is desirable it should be handled via factory parameters that will e.g. set a flag in constructed `DBCordaConsumerImpl` that would be used in filtering.
But, as mentioned above, all implementations should behave in the same way. So if Kafka is filtering out messages with `null` values then DB implementation should do it as well.
Also, when processing DLQ topic, we would most likely want to see original messages, i.e. with original values (e.g. to analyze what caused deserialization error), not messages with `null` values. With current implementation, DLQ processor would probably have to provide no-op serializers.
In case we want to keep messages with null values (in all implementations), we would need to modify serailizers to provide serialization result, not just silently return `null` value.

---

Great points; I think the correct fix here then is to change how the DLQ messages from the serialiser are constructed to prevent them from being null and therefore filtered out (and non-useful...)